Source/Charts/Animation/Animator.swift

```diff
@@ -29,15 +29,65 @@ public protocol AnimatorDelegate
 @objc(ChartAnimator)
 open class Animator: NSObject
 {
+    public enum Dimension: Int {
+        case X, Y, H
+    }
+    
+    public struct State
+    {
+        var phase: Double = 1.0
+        var duration: TimeInterval = 0.0
+        var startTime: TimeInterval = 0.0
+        var endTime: TimeInterval = 0.0
+        var enabled: Bool = false
+        var easing: ChartEasingFunctionBlock?
+        
+        mutating func updatePhase(currentTime: TimeInterval) {
+            var elapsed = currentTime - startTime
+            if elapsed > duration
+            {
+                elapsed = duration
+            }
+            self.phase = easing?(elapsed, duration) ?? Double(elapsed / duration)
+        }
+    }
+    private var animatedDimensions: [Dimension: State] = [:]
+    
     @objc open weak var delegate: AnimatorDelegate?
     @objc open var updateBlock: (() -> Void)?
     @objc open var stopBlock: (() -> Void)?
     
     /// the phase that is animated and influences the drawn values on the x-axis
-    @objc open var phaseX: Double = 1.0
+    @objc open var phaseX: Double {
```

Comment:
The API you're replacing allowed you to set `phaseX`. It's not obvious what the replacement is. Does it make sense to declare three properties like `var xDimension: State` or something of the like? What are your thoughts?

You're already making a fairly large change to the API, so I'm okay making some more. Let's make sure that it makes the most sense before we merge it. I've been wanting to do something the like `State` struct in a lot of different areas of the framework.